Oracle.md

```diff
@@ -195,3 +196,15 @@ The TLV serialization of oracle announcements is as follow.
    * [`oracle_event`:`oracle_event`]
 
 where `signature` is a Schnorr signature over a sha256 hash of the serialized `oracle_event`.
+
+## Signing Algorithm
+
+Signatures should be generated following the algorithm specified in [BIP 340](https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki).
+Tagged hashes should also be used as defined in the [design section of BIP 340](https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki#Design).
+
+The secret key (`sk`) should be the private key corresponding to the oracle's public key.
+
+The algorithm `Sign(sk, message)` is defined as:
+
+* Let `m` = hash<sub>oracle</sub>(message)
```

Comment:
Thanks @benthecarman. We had a short discussion about this last meeting.
There are at least two but potentially more things an oracle can sign for:

1. announcements
2. attestations

So we want to make sure to separate these things. I would have liked to avoid having different tagged hashes these *but* it looks like announcements are necessarily just encoded differently to attestations because:

1. announcements are structured data (so TLVs, JSON etc)
2. attestations are just single strings (I want this to make using oracle signatures easy and useful on their own and not require wonky encoding implementations).

If they were both encoded the same we could just have different markers for each type of message to separate and then a single "oracle" in the tagged hash. But given the different encoding requirements we might do best to push up the application separation into the tag.

I suggest two tags:

1. `oracle/attestation`
2. `oracle/announcement`

and therefore changing the spec here to Sign(sk, message, tag):

* Let H = tag_hash("oracle/" || tag);
* Let m = H(message)
* Return BIP340_sign(sk,m)

and then just invoking `Sign(sk,message,"announcement")` in the announcement part of the spec etc.
Also as we've discussed before it would be better to have an acronym for the specification documents so we can just put `BET04` in there rather than `oracle`.

Reply:
Updated to use `oracle/attestation` & `oracle/announcement`.

> Also as we've discussed before it would be better to have an acronym for the specification documents so we can just put BET04 in there rather than oracle.

Since we don't have a nice acronym for the DLC spec we just do something like `DLCOraclev0/attestation`

Reply:
Good idea. Maybe `DLC/oracle/attestation/v0` since it's a bit clearer?

Reply:
Updated :)